libs/language-server/src/lib/ast/wrappers/value-type/atomic-value-type.ts

```diff
@@ -84,7 +100,7 @@ export class AtomicValueType
 
   protected override doGetSupertype(): ValueType | undefined {
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-    const supertype = this.astNode?.attribute?.type;
+    const supertype = this.astNode?.attribute?.valueType;
     return this.wrapperFactories.ValueType.wrap(supertype);
   }
```

Comment:
Just so we don't forget about it: we should probably get rid of the `supertype` thing and replace it with a better named mechanic